File: prep/implementation/patterns_reference_implementation/bitwise_xor/base10_complement.cpp
#include <algorithm>
#include <cmath>
#include <iostream>

using namespace std;

// O(b) / O(1)
int complement(unsigned int n) {
  int bit_count = 0;
  int num = n;

  while (num) {
    bit_count++;
    num >>= 1;
  }

  int all_set_bits = pow(2, bit_count) - 1;

  return n ^ all_set_bits;
}

int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);

  int tt, n;
  cin >> tt;

  while (tt--) {
    cin >> n;

    cout << complement(n) << "\n";
  }

  return 0;
}

File: prep/implementation/patterns_reference_implementation/bitwise_xor/Base10Complement.java
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;

public class Base10Complement {
  public static void main(String[] args) {
    try (Scanner in = new Scanner(System.in)) {
      int tt = in.nextInt();

      while (tt-- > 0) {
        int n = in.nextInt();
        System.out.println(complement(n));
      }
    }
  }

  // O(b) / O(1)
  public static int complement(int n) {
    int bitCount = 0;
    int num = n;

    while (num > 0) {
      bitCount++;
      num >>= 1;
    }

    int allSetBits = (int)Math.pow(2, bitCount) - 1;

    return n ^ allSetBits;
  }
}